lib/chainsync/chainsync/analysis/calc_pnl.py

```diff
@@ -91,78 +94,40 @@ def calc_single_closeout(position: pd.Series, contract: Contract, pool_info: pd.
         except Exception as exception:  # pylint: disable=broad-except
             logging.warning("Exception caught, ignoring: %s", exception)
 
-    elif tokentype == "LP":
-        fn_args = (
-            amount,
-            min_output,
-            (  # IHyperdrive.Options
-                address,  # destination
-                True,  # asBase
-                bytes(0),  # extraData
-            ),
-        )
-        # If this fails, keep as nan and continue iterating
-        try:
-            preview_result = smart_contract_preview_transaction(
-                contract, sender, "removeLiquidity", *fn_args, block_number=position["block_number"]
-            )
-            out_pnl = Decimal(
-                preview_result["baseProceeds"]
-                # We assume all withdrawal shares are redeemable
-                + preview_result["withdrawalShares"] * pool_info["lp_share_price"].values[-1]
-            ) / Decimal(1e18)
-        except Exception as exception:  # pylint: disable=broad-except
-            logging.warning("Exception caught, ignoring: %s", exception)
-
-    elif tokentype == "WITHDRAWAL_SHARE":
-        fn_args = (
-            amount,
-            min_output,
-            (  # IHyperdrive.Options
-                address,  # destination
-                True,  # asBase
-                bytes(0),  # extraData
-            ),
-        )
-        try:
-            # For PNL, we assume all withdrawal shares are redeemable
-            # even if there are no withdrawal shares available to withdraw
-            # Hence, we don't use preview transaction here
-            out_pnl = Decimal(amount * pool_info["lp_share_price"].values[-1]) / Decimal(1e18)
-        except Exception as exception:  # pylint: disable=broad-except
-            logging.warning("Exception caught, ignoring: %s", exception)
+    elif tokentype in ["LP", "WITHDRAWAL_SHARE"]:
```

Comment:
probably would be helpful to preserve this comment; I was confused for a sec why you didn't have to preview:

```python
# For PNL, we assume all withdrawal shares are redeemable
# even if there are no withdrawal shares available to withdraw
# Hence, we don't use preview transaction here
```